src/lib/convertNumberBase.ts

```diff
@@ -0,0 +1,52 @@
+import BigNumber from 'bignumber.js';
+
+import { NumberBaseType } from './numberBaseConvertProperties';
+
+export function convertNumberBase(
+  value: string,
+  fromUnit: NumberBaseType,
+  toUnit: NumberBaseType,
+): string | undefined {
+  for (const [fromBaseName, fromBaseNumber] of Object.entries(
+    numberBaseConversionDict,
+  )) {
+    if (fromBaseName === fromUnit) {
+      for (const [toBaseName, toBaseNumber] of Object.entries(
+        numberBaseConversionDict,
+      )) {
+        if (toBaseName === toUnit) {
+          return convertNumberBaseWithBn(
+            value,
+            toBaseName,
+            fromBaseNumber,
+            toBaseNumber,
+          );
+        }
+      }
+    }
+  }
+}
+
+// @internal
+const numberBaseConversionDict = {
```

Comment:
I don't like adding Dict at the end of the variable names and we've never done that before. This implies that we would also have XStr, XNumber, XList etc. We've had a convention for naming dicts. Either use some descriptive name or XToY or ByX. `baseToRadix` will do in that case.